### PR TITLE
[BUG] - False warning when using Nextjs next/head to inject partials (Issue/3137)

### PR DESCRIPTION
**Pull request by henrijoss**
### Pull Request Checklist

- [ ] I have read and accept the [Contributor License Agreement](https://opensource.porsche.com/docs/cla)

#### References

- Preview: https://designsystem.porsche.com/issue/3137/partials/introduction

#### Scope

Provide partials as 'js' format in order to use them in Next.js with the Metadata API

#### Resolved Issue

Resolves [#3137](https://github.com/porsche-design-system/porsche-design-system/issues/3137)

### CODE REVIEW

File: packages/components-js/projects/partials/scripts/generateMetaTagsAndIconLinksPartial.ts [hunk 1]
```diff
@@ -15,11 +15,37 @@ const convertToJSX = (templates: string[]): JSX.Element[] => {
 export const generateMetaTagsAndIconLinksPartial = (): string => {
   const metaIconCDNPath = `$cdnBaseUrl/${CDN_BASE_PATH_META_ICONS}`;
 
+  const metadata = `{
+      themeColor: [
+        { media: '(prefers-color-scheme: light)', color: '#FFF' },
+        { media: '(prefers-color-scheme: dark)', color: '#0E1418' },
```
Comment (michaelworm) [inline packages/components-js/projects/partials/scripts/generateMetaTagsAndIconLinksPartial.ts:21]: Would it be possible to import those hex values?

File: packages/components-js/projects/partials/tests/unit/specs/getMetaTagsAndIconLinks.spec.ts [hunk 2]
```diff
@@ -35,16 +35,80 @@ describe('format: jsx', () => {
...
 });
+
+describe('format: js', () => {
+  it('should return meta tags and icon links', () => {
+    const title = 'Porsche UX Platform';
+    const result = getMetaTagsAndIconLinks({ format: 'js', appTitle: title });
+    const expected = {
+      themeColor: [
+        { media: '(prefers-color-scheme: light)', color: '#FFF' },
+        { media: '(prefers-color-scheme: dark)', color: '#0E1418' },
```
Comment (michaelworm) [inline packages/components-js/projects/partials/tests/unit/specs/getMetaTagsAndIconLinks.spec.ts:59]: See above, import, or maybe share the whole config somewhere so it's not tripled

File: packages/components-js/projects/partials/tests/unit/specs/getMetaTagsAndIconLinks.spec.ts [hunk 3]
```diff
@@ -35,16 +35,80 @@ describe('format: jsx', () => {
...
+    expect(result).toMatchObject(expected);
+  });
+
+  it('should return meta tags and icon links for china cdn', () => {
+    const title = 'Porsche UX Platform';
+    const result = getMetaTagsAndIconLinks({ format: 'js', appTitle: title, cdn: 'cn' });
+    const expected = {
+      themeColor: [
+        { media: '(prefers-color-scheme: light)', color: '#FFF' },
+        { media: '(prefers-color-scheme: dark)', color: '#0E1418' },
```
Comment (michaelworm) [inline packages/components-js/projects/partials/tests/unit/specs/getMetaTagsAndIconLinks.spec.ts:90]: See above, import/share

File: packages/components-react/projects/nextjs/app/layout.tsx [hunk 4]
```diff
@@ -1,24 +1,50 @@
...
+
+export const viewport: Viewport = {
+  themeColor,
+};
+
+export const metadata: Metadata = {
+  title: 'Porsche Design System - NextJS',
+  appleWebApp,
+  icons,
+};
```
Comment (michaelworm) [inline packages/components-react/projects/nextjs/app/layout.tsx:26]: Create a const for appTitle/title, so it can be shared

**Review by michaelworm — DISMISSED** [review]

File: packages/components-js/projects/partials/tests/unit/specs/getMetaTagsAndIconLinks.spec.ts (hunk 2 above)
Comment (henrijoss) [reply packages/components-js/projects/partials/tests/unit/specs/getMetaTagsAndIconLinks.spec.ts:59]: For the unit test i would keep it hardcoded so you know when something changes

**Review by michaelworm — APPROVED** [review]

**Pull request merged**